### PR TITLE
Improve `cone_from_inequalities`

### PR DESCRIPTION
**Pull request by alexej-jordan**
Add missing docstrings to the docs for cones.
Extended type support for `AbstractCollection[LinearHalfspace/Hyperplane]`.
Add `cone_from_equations`. This could possibly called by a new method `cone_from_inequalities(::Type{T}, ::Nothing, ::AbstractCollection[LinearHyperplane])`?
This solves #2218 .

### CODE REVIEW

File: src/PolyhedralGeometry/Cone/constructors.jl [hunk 1]
```diff
@@ -118,8 +118,40 @@ function cone_from_inequalities(::Type{T}, I::AbstractCollection[LinearHalfspace
     end
 end
 
+@doc raw"""
+
```
Comment (fingolfin) [inline src/PolyhedralGeometry/Cone/constructors.jl]:
```suggestion
```

File: src/PolyhedralGeometry/Cone/constructors.jl [hunk 2]
```diff
@@ -118,8 +118,40 @@ function cone_from_inequalities(::Type{T}, I::AbstractCollection[LinearHalfspace
...
+    cone_from_equations([::Type{T} = QQFieldElem,] E::AbstractCollection[LinearHyperplane]; non_redundant::Bool = false)
+
+The (convex) cone defined by
+
+$$\{ x | Ex = 0 \}.$$
+
```
Comment (fingolfin) [inline src/PolyhedralGeometry/Cone/constructors.jl]:
````suggestion
```math
\{ x | Ex = 0 \}.
```
````

**Review by YueRen — APPROVED** [review]
Looks good, thank you!

File: test/PolyhedralGeometry/extended.jl [hunk 3]
```diff
@@ -181,6 +181,9 @@
             
             @test cone_from_inequalities(facets(x), collect(affine_hull(y))) == x
             @test cone_from_inequalities(facets(y), collect(linear_span(x))) == x
+
+	    @test cone_from_inequalities([[-1, 0, 0], [0, -1, 0]], [0 0 1]) == x
+	    @test cone_from_inequalities([-1 0 0; 0 -1 0], [[0, 0, 1]]) == x
         end
```
Comment (fingolfin) [inline test/PolyhedralGeometry/extended.jl]: Please avoid tabs
```suggestion

            @test cone_from_inequalities([[-1, 0, 0], [0, -1, 0]], [0 0 1]) == x
            @test cone_from_inequalities([-1 0 0; 0 -1 0], [[0, 0, 1]]) == x
        end
```

File: test/PolyhedralGeometry/Cone.jl [hunk 4]
```diff
@@ -128,5 +128,6 @@ const pm = Polymake
         @test cone_from_inequalities(T, [-1 0 0; 0 0 -1]; non_redundant = true) == Cone2
         @test cone_from_inequalities(T, facets(Cone4), linear_span(Cone4)) == Cone4
         @test cone_from_inequalities(T, facets(Cone4), linear_span(Cone4); non_redundant = true) == Cone4
+	@test cone_from_equations(T, [0 1 0]) == cone_from_inequalities(T, Matrix{Int}(undef, 0, 3), linear_span(Cone4))
```
Comment (fingolfin) [inline test/PolyhedralGeometry/Cone.jl]:
```suggestion
        @test cone_from_equations(T, [0 1 0]) == cone_from_inequalities(T, Matrix{Int}(undef, 0, 3), linear_span(Cone4))
```

**Pull request merged**